File: Java/PrisonerSimulator/src/Game/Decision.java
package Game;

public enum Decision {
    Cooperate,
    Defect;

    public String toString() {
        switch (this) {
            case Cooperate:
                return "Cooperate";
            case Defect:
                return "Defect";
        }
        return "Unknown decision";
    }
}

File: Kotlin/PrisonerSimulator/src/game/Decision.kt
package game

enum class Decision {
    Cooperate {
        override fun getOpposite(): Decision = Defect
        override fun toString(): String = "Cooperate"
    },
    Defect {
        override fun getOpposite(): Decision = Cooperate
        override fun toString(): String = "Defect"
    };

    abstract fun getOpposite(): Decision
}
